src/JestProcessManagement/JestProcessManager.ts

```diff
@@ -31,13 +31,15 @@ export class JestProcessManager {
   } = {}): JestProcess {
     if (watchMode !== WatchMode.None && this.runAllTestsFirstInWatchMode) {
       return this.runAllTestsFirst(exitedJestProcess => {
-        this.removeJestProcessReference(exitedJestProcess)
-        const jestProcessInWatchMode = this.run({
-          watchMode: WatchMode.Watch,
-          keepAlive,
-          exitCallback,
-        })
-        exitCallback(exitedJestProcess, jestProcessInWatchMode)
+        if (!exitedJestProcess.stopRequested()) {
+          this.removeJestProcessReference(exitedJestProcess)
```

Comment:
yeah decided to leave refactoring out, there are other JestProcess related refactoring in our technical debt list, could tackle them at a separate PR

Reply:
In the meantime I would nevertheless move the `removeJestProcessReference` line out of the `if` block. Or am I missing something?

Reply:
If a process is stopped it would have been removed, see the stop()/stopAll() in `JestProcessManager`, therefore, only need to call `removeJestProcessReference` if it is not stopped.

Reply:
Ah thanks, I didn't think of that.